ansible/roles/openshift_udp_limits/tasks/main.yml

```diff
@@ -0,0 +1,172 @@
+  set_fact:
+    cluster_master: "{{ groups['oo_hosttype_master'] | intersect(groups['oo_clusterid_' ~ oo_clusterid]) | first }}"
+  run_once: true
+
+- oc_obj:
+    state: list
+    kind: hostsubnet
+  register: hostsubnets
+  delegate_to: "{{ cluster_master }}"
+  run_once: true
+
+- set_fact:
+    kubernetes_node_name: "{{ (hostvars[inventory_hostname].ec2_private_dns_name if hostvars[inventory_hostname].oo_location == 'aws' else inventory_hostname) }}"
+
+- set_fact:
+    node_cidr: |-
+      {%- for item in hostsubnets['results']['results'][0]['items'] -%}
+       {%- if item.host == kubernetes_node_name -%}
+        {{ item.subnet }}
+       {%- endif -%}
+      {%- endfor -%}
+
+- set_fact:
+    oo_udp_chain:
+    - chain: oo-udp
+      after: "filter$"
+      regex: :oo-udp - \[0:0\]
+      block:  ":oo-udp - [0:0]"
+
+    oo_udp_limit_chain:
+    - chain: oo-udp-limit
+      after: "filter$"
+      regex: :oo-udp-limit - \[0:0\]
+      block:  ":oo-udp-limit - [0:0]"
+
+    filtering_udp_rules:
+    # allow all outbound UDP for privileged pods
+    # send all other traffic to the oo-udp chain so that rules that allow
+    # traffic can call -j RETURN instead of -j ACCEPT, and thus permit us
+    # to do later accounting
+    - chain: OPENSHIFT-OUTBOUND-FILTERING
+      after: "OPENSHIFT-OUTBOUND-FILTERING"
+      regex: "id_outbound_udp_filtering_2_"
+      line: "-I OPENSHIFT-OUTBOUND-FILTERING -m comment --comment id_outbound_udp_filtering_2_ -s {{ node_cidr }} -o eth0 -p udp -j oo-udp"
+
+    - chain: OPENSHIFT-OUTBOUND-FILTERING
+      after: "id_outbound_udp_filtering_2_"
+      regex: "id_outbound_udp_filtering_1_"
+      line: "-I OPENSHIFT-OUTBOUND-FILTERING -m comment --comment id_outbound_udp_filtering_1_ -s 172.0.0.0/8 -o eth0 -p udp -j ACCEPT"
+
+    filtering_udp_memory_rules:
+    - chain: OPENSHIFT-OUTBOUND-FILTERING
+      regex: "id_outbound_udp_filtering_2_"
+      line: "-I OPENSHIFT-OUTBOUND-FILTERING -m comment --comment id_outbound_udp_filtering_2_ -s {{ node_cidr }} -o eth0 -p udp -j oo-udp"
+
+    - chain: OPENSHIFT-OUTBOUND-FILTERING
+      regex: "id_outbound_udp_filtering_1_"
+      line: "-I OPENSHIFT-OUTBOUND-FILTERING -m comment --comment id_outbound_udp_filtering_1_ -s 172.0.0.0/8 -o eth0 -p udp -j ACCEPT"
+
+    outbound_udp_rules:
+    # Allow all DNS traffic to whitelisted DNS servers, with no limits.
+    # Limit any remaining permitted UDP traffic to 30 datagrams per second,
+    # and define permitted UDP traffic for un-privileged pods
+    # Finally, drop all other UDP traffic. Log dropped traffic up to once per minute.
+    # Because all dropped traffic is crammed into this single limited log rule,
+    # log entries will only be a sampling of actual dropped traffic.
+    - chain: oo-udp
+      after: "id_outbound_udp_filtering"
+      regex: "id_outbound_udp_rule_1_"
+      block: |-
+        -A oo-udp -d 172.16.0.0/12 -o eth0 -p udp -m udp --dport 53 -j RETURN
```

Comment:
Nevermind on this, I think it would be better if we have a single rule that allows DNS traffic to the service network, like this:

First, above, you'd fetch the clusternetworks config:
```
  - oc_obj:
      state: list
      kind: clusternetworks
      name: default
    register: clusternetworks
    delegate_to: "{{ cluster_master }}"
    run_once: true
```
then you'd set a fact for the service network:
```
  - set_fact:
      servicenetwork: "{{clusternetworks.results.results[0].serviceNetwork}}"
```
You could combine that new fact with the other `set_fact` task if you want.

Then, your rule here would use that instead of `172.16.0.0/12`:
```
-A oo-udp -d {{servicenetwork}} -o eth0 -p udp -m udp --dport 53 -j RETURN
```